sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1208,36 +1204,72 @@ object SQLContext {
   private val INSTANTIATION_LOCK = new Object()
 
   /**
-   * Reference to the last created SQLContext.
+   * The active SQLContext for threads.
+   */
+  private val activeContexts: InheritableThreadLocal[SQLContext] =
```

Comment:
Nit: I think this should be singular since there is only one active context per thread.